torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -912,15 +912,18 @@ class TORCH_API ProcessGroup : public torch::CustomClassHolder {
   bool hasHooks() const {
     // `getDefaultBackend` will throw today if the backend is set to `undefined`
     // (in case of `init_process_group(nothing)`)
-    try {
-      return getDefaultBackend()->hasHooks();
-    } catch (const std::exception& e) {
+    if (backendTypeToBackend_.find(backendType_) ==
+        backendTypeToBackend_.end()) {
       TORCH_WARN(
-          "Failed to check if ProcessGroup has hooks: ",
-          e.what(),
+          "No backend of type ",
+          uint16_t(backendType_),
+          " found for Process Group with name ",
+          getBackendName(),
           ". Assuming no hooks are registered.");
       return false;
     }
+
+    return getDefaultBackend()->hasHooks();
```

Comment:
This looks okay, but can we directly use the result of `backendTypeToBackend_.find` to avoid a second search?